src/modules/simulator/accelsim/accelsim.cpp

```diff
@@ -318,65 +315,46 @@ ACCELSIM::~ACCELSIM()
 int
 ACCELSIM::init()
 {
-	int ret = -1;
-
-	struct mag_report mrp = {};
-	sensor_accel_s arp = {};
-
 	/* do SIM init first */
-	if (VirtDevObj::init() != 0) {
+	int ret = VirtDevObj::init();
+
+	if (ret != PX4_OK) {
 		PX4_WARN("SIM init failed");
-		goto out;
+		return ret;
 	}
 
 	/* allocate basic report buffers */
 	_accel_reports = new ringbuffer::RingBuffer(2, sizeof(sensor_accel_s));
 
 	if (_accel_reports == nullptr) {
-		goto out;
+		PX4_WARN("_accel_reports creation failed");
+		return -ENOMEM;
 	}
 
-	_mag_reports = new ringbuffer::RingBuffer(2, sizeof(mag_report));
+	_mag_reports = new ringbuffer::RingBuffer(2, sizeof(sensor_mag_s));
 
 	if (_mag_reports == nullptr) {
-		goto out;
+		PX4_WARN("_mag_reports creation failed");
+		return -ENOMEM;
 	}
 
 	/* do init for the mag device node */
 	ret = _mag->init();
 
-	if (ret != OK) {
+	if (ret != PX4_OK) {
 		PX4_WARN("MAG init failed");
-		goto out;
+		return ret;
 	}
 
 	/* fill report structures */
 	_measure();
 
-	/* advertise sensor topic, measure manually to initialize valid report */
-	_mag_reports->get(&mrp);
-
-	/* measurement will have generated a report, publish */
-	_mag->_mag_topic = orb_advertise_multi(ORB_ID(sensor_mag), &mrp,
-					       &_mag->_mag_orb_class_instance, ORB_PRIO_LOW);
+	sensor_mag_s mag_report = {};
 
-	if (_mag->_mag_topic == nullptr) {
-		PX4_WARN("ADVERT ERR");
-	}
+	/* advertise mag sensor topic, measure manually to initialize valid report */
+	_mag_reports->get(&mag_report);
```

Comment:
You're right, I'm pretty sure you can remove all these lines:
```
	/* fill report structures */
	_measure();
	sensor_mag_s mag_report = {};
	/* advertise mag sensor topic, measure manually to initialize valid report */
	_mag_reports->get(&mag_report);
```

The rest looks good.

Reply:
Done and pushed up.  I deleted the measure() call from gyrosim.cpp init() to keep the two files/methods aligned in the most recent commit. If it was a misstep to take it out of gyrosim.cpp init() just let me know!  Thank you!